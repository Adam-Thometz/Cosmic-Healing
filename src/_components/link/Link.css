.Link {
  font-size: 16px;
  font-weight: 700;
  text-align: center;
  cursor: pointer;
  border-radius: 4px;
  font-family: Nuckle;
}

.Link-arrow {
  margin: 8px 0 8px 8px;
}

.primary {
  background-color: #B279F1;
  color: #FFFFFF;
  padding: 8px 16px;
  box-shadow: 0px 4px 35px 0px #782DCA26;
  transition: background-color 0.3s;
  width: 150px;
}

.secondary {
  display: flex;
  align-items: center;
  background-color: rgba(0,0,0,0);
  border: 2px solid black;
  padding: 4px 16px;
  width: 110px;
  font-weight: 700;
}

.primary:hover{
  background-color: #8F59CB;
}

.secondary:hover {
  background-color: #FFC5A21E;
  border-color: #D87B41;
  filter: invert(47%) sepia(71%) saturate(466%) hue-rotate(339deg) brightness(102%) contrast(86%);
}

.tertiary {
  text-decoration: none;
  color: white;
  font-weight: 600;
  margin: 8px 0;
  text-align: left;
}

.tertiary:hover {
  color: #D87B41;
}