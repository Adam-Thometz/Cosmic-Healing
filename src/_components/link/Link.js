import React from 'react';

import './Link.css';

import arrow from './arrow.png';

const Link = ({ primary = false, secondary = false, link, children }) => {
  const btnClass = primary ? 'primary' : (
    secondary ? 'secondary' : 'tertiary'
  );
  
  return <a href={link} className={`Link ${btnClass}`}>
    {children}
    {secondary ? <img className='Link-arrow' src={arrow} alt='' /> : null}
  </a>;
};

export default Link;