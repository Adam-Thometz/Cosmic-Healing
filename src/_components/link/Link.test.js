import React from "react";
import { render } from "@testing-library/react";
import Link from './Link';

describe('Link component', () => {
  const component = <Link primary>Hello, I'm a primary link</Link>;
  const component2 = <Link>Hello, I'm a regular link</Link>;
  it('renders without crashing', () => {
    render(component);
  });

  it('matches the snapshot for primary link', () => {
    const { asFragment } = render(component);
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches the snapshot for secondary link', () => {
    const { asFragment } = render(component2);
    expect(asFragment()).toMatchSnapshot();
  });
});