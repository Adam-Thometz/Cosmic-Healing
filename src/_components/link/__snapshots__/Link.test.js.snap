// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link component matches the snapshot for primary link 1`] = `
<DocumentFragment>
  <a
    class="Link primary"
  >
    Hello, I'm a primary link
  </a>
</DocumentFragment>
`;

exports[`Link component matches the snapshot for secondary link 1`] = `
<DocumentFragment>
  <a
    class="Link tertiary"
  >
    Hello, I'm a regular link
  </a>
</DocumentFragment>
`;
